benches/suites/raw_node.rs

```diff
@@ -20,3 +25,107 @@ pub fn bench_raw_node_new(c: &mut Criterion) {
 
     c.bench_function("RawNode::new", bench);
 }
+
+pub fn bench_raw_node_leader_propose(c: &mut Criterion) {
+    static KB: usize = 1024;
+    let mut test_sets = vec![
+        0,
+        32,
+        128,
+        512,
+        KB,
+        4 * KB,
+        16 * KB,
+        128 * KB,
+        512 * KB,
+        KB * KB,
+    ];
+    let mut group = c.benchmark_group("RawNode::leader_propose");
+    for size in test_sets.drain(..) {
+        // Calculate measurement time in seconds according to the input size.
+        // The approximate time might not be the best but should work fine.
+        let mtime = if size < KB {
+            1
+        } else if size < 128 * KB {
+            3
+        } else {
+            7
+        };
+        group
+            .measurement_time(Duration::from_secs(mtime))
+            .throughput(Throughput::Bytes(size as u64))
+            .bench_with_input(
+                BenchmarkId::from_parameter(size),
+                &size,
+                |b: &mut Bencher, size| {
+                    let logger = crate::default_logger();
+                    let mut node = quick_raw_node(&logger);
+                    node.raft.become_candidate();
+                    node.raft.become_leader();
+                    b.iter_batched(
+                        || (vec![0; 8], vec![0; *size]),
+                        |(context, value)| node.propose(context, value).expect(""),
+                        BatchSize::SmallInput,
+                    );
+                },
+            );
+    }
+}
+
+pub fn bench_raw_node_new_ready(c: &mut Criterion) {
+    let logger = crate::default_logger();
+    let mut group = c.benchmark_group("RawNode::ready");
+    group
+        // TODO: The proper measurement time could be affected by the system and machine.
+        .measurement_time(Duration::from_secs(60))
+        .bench_function("Default", |b: &mut Bencher| {
+            b.iter_batched(
+                || test_ready_raft_node(&logger),
+                |mut node| {
+                    let _ = node.ready();
```

Comment:
From reading https://github.com/bheisler/criterion.rs/blob/master/book/src/faq.md#when-should-i-use-criterionblack_box I think we don't need to use it here. :)

Reply:
Do you mean the macro black_box? I haven’t used it in our benches.

Reply:
Yes I think the code is fine, I just was investigating for myself to make sure we didn't need to consider it. Good job. :)